### PR TITLE
Use importlib instead of deprecated imp module

### LINKED ISSUE #234: DeprecationWarning for 'imp' package
Hello,
I'm running the 0.4.2 version of `skfuzzy` and have recently started seeing the following warning:
```DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
```
This happens on the latest python `3.7` version. Thought I'd give you a heads-up as presumably this import will break at some point in the future.

### PR DESCRIPTION
**Pull request by kinow**
Tentative fix, closes #234 

The Python 3 [docs](https://docs.python.org/3/library/imp.html#imp.find_module) appears to suggest using other methods to replace `find_module`, but given the code is just trying to import `nose` or doing something else in case it fails, I think `importlib.import_module` should work?

```py
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import importlib
>>> importlib.import_module('nose')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kinow/Development/python/workspace/scikit-fuzzy/venv/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'nose'
>>> 
```

The `ModuleNotFoundError` extends `ImportError`, so the `except` statement should run fine if `nose` is not found I think.

### CODE REVIEW

**Comment by JDWarner** [comment]
This works fine under 3.x, but fails in 2.x as the API wasn't available there previously.

Putting the `imp` into a `try:` conditional with this new method in the `except` should make it backwards-compatible and make the tests happy.

**Comment by kinow** [comment]
Sorry, didn't think about py 2.x. Will update this PR today and test with Ubuntu LTS' py2 too.

**Comment by kinow** [comment]
Hi @JDWarner 

>Putting the imp into a try: conditional with this new method in the except should make it backwards-compatible and make the tests happy.

I've updated the code, but did the other way. `importlib` into a try, and `imp` in the except. It should still be backward compatible, with no exception raised for the user. Importing `importlib` first should fix #234, as users with `nose` & `importlib` won't have a warning any longer.

Does that make sense? Happy to revert and use your suggestion if you prefer.

Also almost forgot to update the `del` statement at the end of the file. Tried putting it now in a `finally` block that checks if the modules were successfully imported first. Not sure if the best way?

Thanks!

File: skfuzzy/__init__.py [hunk 1]
```diff
@@ -71,14 +70,21 @@
...
+        # py2
+        import imp
+        imp.find_module('nose')
+        del imp
+    except ImportError:
+        def _test(doctest=False, verbose=False):
+            """This would run all unit tests, but nose couldn't be
+            imported so the test suite can not run.
+            """
+            raise ImportError("Could not load nose. Unit tests not available.")
```
Comment (JDWarner) [inline skfuzzy/__init__.py]: The remaining error is due to this new nesting not defining `_test()` before it is used in line 119 in Python 2; previously this was defined in the `else` of the `try-except-else` after the exception clause isn't triggered (Python 2 branch).

Under the new structure in Python 2, the top-level `else` does not execute and `_test` is not defined when used.  Simplified example:

``` python
try:
    print("first try statement")
    # force Py2 path
    import notamodule
except ImportError:
    try:
        print("python 2, nose available")
        # import notamodule  # Uncomment to simulate Py2 without Nose available (which works)
    except ImportError:
        print("_test defined")
else:
    print("_test defined")
finally:
    print("cleanup")
```

One solution - though perhaps not the most elegant - is to move the definition of `_test` into the `finally` clause and make it conditional on `_test` not existing or with a try/except `NameError`.  

Could also have both different `_test` functions defined at the top level and use `test = _test_withnose` or `test = _test_nonose` in these structures, which might be the most elegant.

Comment (kinow) [reply skfuzzy/__init__.py]: Ah! Good spot! Sorry. Updated PR using the second option :+1: Thanks!

File: skfuzzy/__init__.py [hunk 2]
```diff
@@ -62,50 +62,68 @@
...
 # Enable testing of the package
 import os.path as osp
-import imp
 import functools
 import warnings
 import sys
 
 pkg_dir = osp.abspath(osp.dirname(__file__))
 data_dir = osp.join(pkg_dir, 'data')
+nose_installed = False
```
Comment (kinow) [inline skfuzzy/__init__.py]: We assume `nose` is not installed by default, and then only try to import for Py3 and Py2. Upon success, we set this flag to `True`. Then later we use this flag to choose which function to use.

**Comment by wouter-vdb** [comment]
Superseded by #265.

**Comment by Nieju** [comment]
Hey @JDWarner , looks like this isn't much more work. Could you maybe include it in a future release?

**Comment by Nieju** [comment]
Would love to see this completed :)

**Comment by JDWarner** [comment]
Superseded by #265

**Pull request closed without merging**